src/RequestInformation.php

```diff
@@ -254,13 +254,7 @@ public function setPathParameters(array $pathParameters): void {
      */
     public function addHeaders(array $headers): void
     {
-        foreach ($headers as $key => $headerValue) {
-            if (is_array($headerValue)) {
-                $this->headers->putAllToKey($key, $headerValue);
-            } else{
-                $this->headers->add($key, strval($headerValue));
-            }
```

Comment:
Moved these checks to `putAll` to allow other modules to easily re-use this e.g. setting headers on a `BatchRequestItem`